ambry-network/src/main/java/com/github/ambry/network/http2/Http2NetworkClient.java

```diff
@@ -144,6 +174,12 @@ public void operationComplete(ChannelFuture future) throws Exception {
     if (requestsToDrop.size() != 0) {
       logger.warn("Number of requestsToDrop: {}", requestsToDrop.size());
       http2ClientMetrics.http2RequestsToDropCount.inc(requestsToDrop.size());
+      for (int correlationId : requestsToDrop) {
+        Channel streamChannel = correlationIdInFlightToChannelMap.remove(correlationId);
+        if (streamChannel != null) {
+          releaseAndCloseStreamChannel(streamChannel);
+        }
+      }
     }
 
     http2ClientResponseHandler.getResponseInfoQueue().poll(readyResponseInfos, pollTimeoutMs);
```

Comment:
is there a possibility that redyResponseInfos contain responses for the dropped request? if so, do we care?

Reply:
It should include responses from dropped request. That's how router knows it was drooped.